lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -2021,9 +2010,19 @@ def _current_config(self) -> dict:
         Returns:
             A dict containing Promtail configuration.
         """
-        raw_current = self._container.pull(WORKLOAD_CONFIG_PATH).read()
-        current_config = yaml.safe_load(raw_current)
-        return current_config
+        if not self._container.can_connect():
+            logger.debug("Could not connect to promtail container!")
+            return {}
+        try:
+            raw_current = self._container.pull(WORKLOAD_CONFIG_PATH).read()
+            return yaml.safe_load(raw_current)
+        except (ProtocolError, PathError) as e:
```

Comment:
Suppose the Loki provider has just received its first relation departed event i.e. for its last LogProxyConsumer. Now suppose that this `_current_config()` method returns an empty dictionary because it got a Protocol Error or Path Error.

In this case I presume the new config should also be an empty dictionary. Hence both current config and new config will be the same when processing the relation departed handler. Is this not a problem ?

Reply:
The new config (or the barebones template from `_promtail_config`) is set [here](https://github.com/canonical/loki-k8s-operator/blob/1152716c1198833aa7be40dbc7dba6d30d0ca371/lib/charms/loki_k8s/v0/loki_push_api.py#L2028), and will be at least:

```python
{
  "clients": [],
   "server": {
      "http_listen_port": HTTP_LISTEN_PORT,
      "grpc_listen_port": GRPC_LISTEN_PORT,
   },
   "positions": {
     "filename": WORKLOAD_POSITIONS_PATH
   },
   "scrape_configs": []
}
```

The comments always bear consideration, though.